guides/intro_to_keras_for_researchers.py

```diff
@@ -705,11 +705,64 @@ def call(self, inputs, training=None):
 Functional models.
 
 Note that the `Model` class also features built-in training & evaluation loops
-(`fit()` and `evaluate()`). You can always subclass the `Model` class
-(it works exactly like subclassing `Layer`) if you want to leverage these loops
-for your OO models.
+(`fit()` and `evaluate()`). You can implement the model for MNIST dataset above
+with the following code.
 """
 
+inputs = tf.keras.Input(shape=(784,), dtype="float32")
+x = keras.layers.Dense(32, activation="relu")(inputs)
+x = keras.layers.Dense(32, activation="relu")(x)
+outputs = keras.layers.Dense(10)(x)
+model = tf.keras.Model(inputs, outputs)
+
+# Specifying the loss, optimizer, and metrics with `compile()`.
+model.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=keras.optimizers.Adam(learning_rate=1e-3),
+    metrics=[keras.metrics.SparseCategoricalAccuracy()])
+
+# Train the model with the dataset for 2 epochs.
+model.fit(dataset, epochs=2)
+model.evaluate(dataset)
+
+"""
+You can always subclass the `Model` class (it works exactly like subclassing
+`Layer`) if you want to leverage these loops for your OO models.  You can
```

Comment:
if you want to leverage built-in training infrastructure for your own training loop, such as:

- Callbacks (link)
- Distributed training support (link)
- Step fusing (link)

Reply:
Let's define these features in the section above and mention them again here